core/storage/trie/polkadot_trie_db/polkadot_trie_db.cpp

```diff
@@ -44,12 +45,12 @@ namespace kagome::storage::merkle {
 
     if (value.empty()) {
       OUTCOME_TRY(remove(key));
-    } else if (not root_.has_value()) {
-      // will create a leaf node with provided key and value, save it to the
-      // storage and return the key to it
-      OUTCOME_TRY(insertRoot(k_enc, value));
     } else {
-      OUTCOME_TRY(root, retrieveNode(root_.value()));
+      NodePtr root = nullptr;
+      if(root_.has_value()) {
+        OUTCOME_TRY(r, retrieveNode(root_.value()));
+        root = r;
```

Comment:
What is the purpose of creation `root` in the `else` scope and assigning to it? It gets deleted on the next line

Reply:
Do you mean  NodePtr root? Why does it? It exists to the end of else block and is actively used in it.